src/layers/data_type_layer.cpp

```diff
@@ -286,10 +286,13 @@ auto data_type_layer<TensorDataType>::get_local_error_signals(int parent_index)
 // Accessing matrices corresponding to parent/child layer
 template <typename TensorDataType>
 auto data_type_layer<TensorDataType>::get_activations(const Layer& child) const -> const BaseDistMat& {
-  const int child_index = (std::find(m_child_layers.begin(),
-                                     m_child_layers.end(),
-                                     &child)
-                           - m_child_layers.begin());
+  if(m_child_layers.empty()) {
+    LBANN_ERROR("This layer has no children");
+  }
+  const int child_index = std::distance(m_child_layers.begin(),
```

Comment:
This should change to a call to `Layer::find_layer_index`, which should be renamed to `find_child_layer_index` and it should return a `size_t` (technically the `difference_type` for whatever iterators are going to be used).

Reply:
The `size_t` part is getting pushed to its own PR.